pandas/core/arraylike.py

```diff
@@ -97,6 +97,54 @@ def _arith_method(self, other, op):
 
     @unpack_zerodim_and_defer("__add__")
     def __add__(self, other):
+        """
+        Get addition of DataFrame and other, column-wise.
+
+        This is equivalent to df.add(other, axis=1).
+
+        Parameters
+        ----------
+        other : array-like
```

Comment:
For the examples, `other` can be at least a scalar, a `Series` and a `DataFrame`. I guess we support the same here as in `add`, you can see the type of  `other` there in its docstring: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.add.html

Reply:
Great. Thanks! Fixed typing and added at least one example for each data type.